src/osd/PrimaryLogPG.cc

```diff
@@ -7964,6 +8063,21 @@ void PrimaryLogPG::make_writeable(OpContext *ctx)
       if (pool.info.is_erasure())
 	ctx->clone_obc->attr_cache = ctx->obc->attr_cache;
       snap_oi = &ctx->clone_obc->obs.oi;
+
+      if (ctx->obc->obs.oi.has_manifest()) {
+	if ((ctx->obc->obs.oi.flags & object_info_t::FLAG_REDIRECT_HAS_REFERENCE) && 
+	    ctx->obc->obs.oi.manifest.is_redirect()) {
+	  snap_oi->set_flag(object_info_t::FLAG_MANIFEST);
+	  snap_oi->manifest.type = object_manifest_t::TYPE_REDIRECT;
+	  snap_oi->manifest.redirect_target = ctx->obc->obs.oi.manifest.redirect_target;
+	} else if (ctx->obc->obs.oi.manifest.is_chunked()) {
+	  snap_oi->set_flag(object_info_t::FLAG_MANIFEST);
+	  snap_oi->manifest.type = object_manifest_t::TYPE_CHUNKED;
+	  snap_oi->manifest.chunk_map = ctx->obc->obs.oi.manifest.chunk_map;
```

Comment:
So, if parts of the existing object were dirty, the clone and the head now have chunk_maps with dirty pointers to the same backing objects?  Not necessarily a problem, just clarifying.

Reply:
object_info_t exists on the upper tier.  So, the head will have dirty pointers if existing object is dirty and the clone will have previous chunk_map (not dirty).

Reply:
If the chunk_map previously was dirty, what happens?  Wouldn't the clone have a dirty chunk_map?

Reply:
Right. the clone would have a dirty chunk_map.